devlib/utils/ssh.py

```diff
@@ -367,25 +367,32 @@ def __init__(self,
         else:
             logger.debug('Using SFTP for file transfer')
 
-        self.client = self._make_client()
-        atexit.register(self.close)
-
-        # Use a marker in the output so that we will be able to differentiate
-        # target connection issues with "password needed".
-        # Also, sudo might not be installed at all on the target (but
-        # everything will work as long as we login as root). If sudo is still
-        # needed, it will explode when someone tries to use it. After all, the
-        # user might not be interested in being root at all.
-        self._sudo_needs_password = (
-            'NEED_PASSWORD' in
-            self.execute(
-                # sudo -n is broken on some versions on MacOSX, revisit that if
-                # someone ever cares
-                'sudo -n true || echo NEED_PASSWORD',
-                as_root=False,
-                check_exit_code=False,
+        self.client = None
+        try:
+            self.client = self._make_client()
+            atexit.register(self.close)
```

Comment:
unrelated to this PR but we should probably fix this: if we keep creating connection objects (and we do because of the multithreading support), we end up:
* accumulating atexit handlers without ever removing them
* leaking the connection object, as a reference to it will stay alive in the method's `__self__`

let me know if you want to get your hands dirty in low level Python, otherwise I'll do it

Reply:
Will leave up to you :)